addons/binding/org.openhab.binding.wso2iots/src/main/java/org/openhab/binding/wso2iots/handler/wso2iotsHandler.java

```diff
@@ -241,41 +203,32 @@ private void updateChannel(String channelId, BigDecimal iotsResponse) {
     private BigDecimal updateBuildingMonitorData(String sensorType) throws Exception {
 
         BigDecimal iotsResponse = null;
-        boolean resultok = false;
         String errormsg = null;
         String response1 = null;
         wso2iotsConfiguration config = getConfigAs(wso2iotsConfiguration.class);
         String Access_Token = config.apikey;
         String id = config.deviceId;
-        // String Access_Token = "bfe52f8a-5f34-382b-802c-93e52468564f";
-        // String id = "2940205";
         long time = System.currentTimeMillis() / 1000;
-        String fromTime = String.valueOf(time - 200);
+        String fromTime = String.valueOf(time - PERIOD);
         String toTime = String.valueOf(time);
 
         String url = "https://localhost:8243/senseme/device/1.0.0/stats/";
         url = url + id + "?from=" + fromTime + "&to=" + toTime + "&sensorType=" + sensorType;
 
         URL obj;
-        // if(Boolean.parseBoolean(System.getProperty(""))) {
-        obj = this.validateCertificate(url);
-
-        HttpURLConnection con = (HttpURLConnection) obj.openConnection();
-
-        // optional default is GET
-        con.setRequestMethod("GET");
+        
+        try {
```

Comment:
For validateCertificate, moving try before it does not change anything, because you catch the exceptions inside that method. Question remains valid - can validateCertificate fail and if yes, what are the consequences to the code following it - will it fail? How will it fail?